cli/internal/cache/cache_fs.go

```diff
@@ -38,21 +42,27 @@ func (f *fsCache) Fetch(target, hash string, _unusedOutputGlobs []string) (bool,
 
 func (f *fsCache) Put(target, hash string, duration int, files []string) error {
 	g := new(errgroup.Group)
-	for i, file := range files {
-		_, file := i, file // https://golang.org/doc/faq#closures_and_goroutines
-		hash := hash
+
+	var numDigesters int = runtime.NumCPU()
+	var totalFiles int = len(files)
+	var chunkSize int = int(math.Ceil(float64(len(files)) / float64(numDigesters)))
+
+	for i := 0; i < numDigesters; i++ {
+		batch := files[chunkSize*i : int(math.Min(float64(chunkSize*i+chunkSize), float64(totalFiles)))]
```

Comment:
What about using a queue or channel rather than fixed batches? That might result in a more even workload between the various workers.

Reply:
@gsoltis This is my first time writing in go so I must admit I didn't think about that. Let me check that out.